geoh5py/objects/surveys/electromagnetics/tipper.py

```diff
@@ -0,0 +1,220 @@
+        "Gigahertz (GHz)",
+    ]
+    __INPUT_TYPE = ["Rx and base stations"]
+    _base_stations = None
+
+    def __init__(
+        self,
+        object_type: ObjectType,
+        base_stations: TipperBaseStations | None = None,
+        **kwargs,
+    ):
+        self._base_stations = base_stations
+
+        super().__init__(
+            object_type,
+            **kwargs,
+        )
+
+    # def copy(
+    #     self, parent=None, copy_children: bool = True
+    # ) -> TipperBaseStations | TipperReceivers:
+    #     """
+    #     Function to copy a TipperReceivers or TipperBaseStations to a different parent entity.
+    #
+    #     :param parent: Target parent to copy the entity under. Copied to current
+    #         :obj:`~geoh5py.shared.entity.Entity.parent` if None.
+    #     :param copy_children: Create copies of AirborneTEMReceivers
+    #         or TipperBaseStations along with the copy.
+    #
+    #     :return entity: Registered BaseTipper to the workspace.
+    #     """
+    #     if parent is None:
+    #         parent = self.parent
+    #
+    #     omit_list = ["_metadata", "_receivers", "_base_stations"]
+    #     new_entity = parent.workspace.copy_to_parent(
+    #         self, parent, copy_children=copy_children, omit_list=omit_list
+    #     )
+    #     if isinstance(new_entity, TipperReceivers):
+    #         new_base_stations = parent.workspace.copy_to_parent(
+    #             self.base_stations,
+    #             parent,
+    #             copy_children=copy_children,
+    #             omit_list=omit_list,
+    #         )
+    #         new_entity.base_stations = new_base_stations
+    #     else:
+    #         new_receivers = parent.workspace.copy_to_parent(
+    #             self.receivers,
+    #             parent,
+    #             copy_children=copy_children,
+    #             omit_list=omit_list,
+    #         )
+    #         new_entity.receivers = new_receivers
+    #
+    #     new_entity.metadata = self.metadata
+    #     parent.workspace.finalize()
+    #
+    #     return new_entity
+
+    @property
+    def base_stations(self) -> TipperBaseStations | None:
+        """The base station entity"""
+        if isinstance(self, TipperBaseStations):
+            return self
+
+        if getattr(self, "_base_stations", None) is None:
+            if (
+                self.metadata is not None
+                and "Base stations" in self.metadata["EM Dataset"]
+            ):
+                tx_uid = self.metadata["EM Dataset"]["Base stations"]
+
+                try:
+                    self._base_stations = self.workspace.get_entity(tx_uid)[0]
+                except IndexError:
+                    warnings.warn(
+                        "Associated `base_stations` entity not set.", UserWarning
+                    )
+                    return None
+
+        return self._base_stations
+
+    @base_stations.setter
+    def base_stations(self, base: TipperBaseStations):
```

Comment:
Add a validation (raise/error) if n_vertices of base stations not one or equal to n_vertices of receivers

Reply:
Done